src/events.c

```diff
@@ -328,7 +328,11 @@ int gg_session_init_ssl(struct gg_session *gs)
 	}
 
 	if (gs->ssl_ctx == NULL) {
-		gs->ssl_ctx = SSL_CTX_new(SSLv3_client_method());
+#ifdef GG_CONFIG_HAVE_TLS_CLIENT_METHOD
+		gs->ssl_ctx = SSL_CTX_new(TLS_client_method());
+#else
+		gs->ssl_ctx = SSL_CTX_new(TLSv1_client_method());
```

Comment:
TLSv1_2_client_method?

If we want to be compatible all the way to TLSv1_client_method, we should include checks for 1.1 and 1.2 as well. And optionally could include a fallback of last resort for SSLv3.

Having said that, I'm fine with leaving just a fallback for TLSv1.2.

Reply:
I assumed that with TLSv1_client_method() was a wildcard for all TLSv1.x protocols, my bad. To use TLSv1_2_client_method() as an only fallback we'd have to bump OpenSSL version requirement from 0.9.7 to 0.9.8 and I'd like to avoid that in a maintenance release.